net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/ApplicationStartedHandler.java

```diff
@@ -0,0 +1,133 @@
+//============================================================================
+//
+// Copyright (C) 2003-2023 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+//
+//============================================================================
+
+package net.sf.eclipsecs.ui;
+
+import org.osgi.service.component.annotations.Component;
```

Comment:
@Bananeweizen My Eclipse is telling me that this import cannot be resolved. Any ideas?

Reply:
![grafik](https://github.com/user-attachments/assets/ba3e4661-5577-4089-a856-4ea947b5c771)

That package is part of the org.eclipse.osgi.services bundle, which in turn is part of our target platform. The most probable reason for you not seeing it would be that you develop against the running IDE application, not against the target platform. Enable Preferences > Plugin Development > Show current target platform in status bar, so you can always easily see what you are developing against. Then open the *.target file and hit "Set as active target platform".

Reply:
I have the correct target platform.
The only way I have found to solve this is to add `org.osgi.service.component.annotations` to the `Import-Package` section of `net.sf.eclipsecs.ui/META-INF/MANIFEST.MF`.

```
--- a/net.sf.eclipsecs.ui/META-INF/MANIFEST.MF
+++ b/net.sf.eclipsecs.ui/META-INF/MANIFEST.MF
@@ -68,6 +68,7 @@ Import-Package: org.apache.commons.lang3;version="3.12.0",
  org.eclipse.ui.texteditor,
  org.eclipse.ui.views.markers,
  org.osgi.framework,
+ org.osgi.service.component.annotations,
  org.osgi.service.event;version="[1.4.0,2.0.0)",
  org.osgi.service.prefs
 Automatic-Module-Name: net.sf.eclipsecs.ui
```

Should I commit this?

Reply:
That would be okay, since it's not wrong. You could try https://github.com/checkstyle/eclipse-cs/pull/831 first, where I explicitly added another bundle to the target platform. I noticed that the relevant package is exported by multiple different bundles in different versions. And maybe that's somehow related to the error that you see: 
![grafik](https://github.com/user-attachments/assets/8bb2255d-2b47-480c-be81-9611adaf4c79)